.github/workflows/run_tests.yml

```diff
@@ -9,7 +9,23 @@ env:
   FORCE_COLOR: 3
 
 jobs:
+  format:
```

Comment:
Q: what's the rationale for making this a separate job / what goes wrong with keeping it in one job?

Reply:
I had a few motivations.
1. Separation of concerns
2. Fast failure when there's a problem. Pre-commit doesn't require the package to be installed and caches the environment, so it spins up faster and fails fast if there's an issue.
3. Minimizing CI usage. Instead of failing 4 jobs when there's a linting error, only one job fails.
4. There's now 2 types of CI tests, the standard and the oldest-supported. Now all tests start with the pre-commit.

Reply:
This makes sense to me. I like the justification.